core/src/main/scala/kafka/log/OffsetIndex.scala

```diff
@@ -132,15 +132,30 @@ class OffsetIndex(_file: File, baseOffset: Long, maxIndexSize: Int = -1, writabl
     }
   }
 
+  /**
+    * Convert given offset to base-relative form. This function throws InvalidOffsetException if the conversion results
+    * in overflow or underflow.
+    * @param offset Offset to convert
+    * @return Base-relative offset
+    * @throws InvalidOffsetException
+    */
+  def getRelativeOffset(offset: Long): Int = {
```

Comment:
This should probably be just `relativeOffset`. We generally avoid the `get` prefix in Kafka.

Reply:
Also, is there a reason why we can't have this and the TimeIndex implementation in AbstractIndex?

Reply:
I thought about adding this to AbstractIndex but it seemed to break the generic abstraction. `relativeOffset` may not be relevant for all kinds of AbstractIndex implementations.

Reply:
Since we have no plans to add any other index types, I think that's fine. We can refactor later if we ever have a need for that.